system-tests/management-api/management-api-test-runner/src/test/java/org/eclipse/edc/test/e2e/managementapi/TransferProcessApiEndToEndTest.java

```diff
@@ -243,6 +243,48 @@ void request_byState() throws JsonProcessingException {
             assertThat(result).anySatisfy(it -> assertThat(it.asJsonObject().getString("state")).isEqualTo(state.toString()));
         }
 
+        @Test
+        void request_sortByStateTimestamp() throws JsonProcessingException, InterruptedException {
+            var tp1 = createTransferProcessBuilder("test-tp1").build();
+            var tp2 = createTransferProcessBuilder("test-tp2").build();
+            getStore().save(tp1);
+            Thread.sleep(3000);
```

Comment:
I would avoid the sleep here, to change the stateTimestamp a mocked `Clock` could be passed to the `TransferProcess`

Reply:
i should not have left that in :(